core/templates/services/math-interactions.service.spec.ts

```diff
@@ -205,6 +214,77 @@ describe('MathInteractionsService', () => {
     expect(mathInteractionsService.getWarningText()).toBe(
       'Your answer contains an invalid term: 3.4.5');
 
+    expect(mathInteractionsService.validateAlgebraicExpression(
+      'a4', ['a'])).toBeFalse();
+    expect(mathInteractionsService.getWarningText()).toBe(
+      'Your answer contains invalid variable multiplication: a4. ' +
+      'The variable should come after the number: 4a.');
+
+    expect(mathInteractionsService.validateAlgebraicExpression(
+      'a * 4', ['a'])).toBeFalse();
+    expect(mathInteractionsService.getWarningText()).toBe(
+      'Your answer contains invalid variable multiplication: a*4. ' +
+      'The variable should come after the number: 4*a.');
+
+    expect(mathInteractionsService.validateAlgebraicExpression(
+      'x^y^z', ['x', 'y', 'z'])).toBeFalse();
+    expect(mathInteractionsService.getWarningText()).toBe(
+      'Your expression contains an exponent in an exponent.');
+
+    expect(mathInteractionsService.validateAlgebraicExpression(
+      '3^4^5', ['a', 'x', 'y', 'z'])).toBeFalse();
+    expect(mathInteractionsService.getWarningText()).toBe(
+      'Your expression contains an exponent in an exponent.');
+
+    expect(mathInteractionsService.validateAlgebraicExpression(
+      'x^(y^z)', ['x', 'y', 'z'])).toBeFalse();
+    expect(mathInteractionsService.getWarningText()).toBe(
+      'Your expression contains an exponent in an exponent.');
+
+    expect(mathInteractionsService.validateAlgebraicExpression(
+      'x^(y*a^z)', ['a', 'x', 'y', 'z'])).toBeFalse();
+    expect(mathInteractionsService.getWarningText()).toBe(
+      'Your expression contains an exponent in an exponent.');
+
+    expect(mathInteractionsService.validateAlgebraicExpression(
+      'a^6', ['a'])).toBeFalse();
+    expect(mathInteractionsService.getWarningText()).toBe(
+      'Your expression contains an exponent with value greater than 5.');
+
+    expect(mathInteractionsService.validateAlgebraicExpression(
+      'a^(3+4)', ['a'])).toBeFalse();
+    expect(mathInteractionsService.getWarningText()).toBe(
+      'Your expression contains an exponent with value greater than 5.');
+
+    expect(mathInteractionsService.validateNumericExpression(
+      '5^100')).toBeFalse();
+    expect(mathInteractionsService.getWarningText()).toBe(
+      'Your expression contains an exponent with value greater than 5.');
+
+    expect(mathInteractionsService.validateAlgebraicExpression(
+      'a+((b-c)+(d+e))', ['a', 'b', 'c', 'd', 'e'])).toBeFalse();
+    expect(mathInteractionsService.getWarningText()).toBe(
+      'Your expression contains redundant parentheses: (b-c).'
+    );
+
+    expect(mathInteractionsService.validateAlgebraicExpression(
+      '-(a + b) + (a + (-b))', ['a', 'b'])).toBeFalse();
+    expect(mathInteractionsService.getWarningText()).toBe(
+      'Your expression contains redundant parentheses: (a+(-b)).'
+    );
+
+    expect(mathInteractionsService.validateAlgebraicExpression(
```

Comment:
Same for this one -- I don't think it should trigger.

Reply:
Fixed.